consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -576,25 +581,29 @@ impl<
         // Purge transactions that can no longer be processed based on their tombstone
         // block.
         let max_externalized_slots = self.scp_node.max_externalized_slots() as u64;
-        let purged_hashes = {
-            let index = self
-                .current_slot_index
-                .saturating_sub(max_externalized_slots);
-            self.tx_manager.remove_expired(index)
-        };
+        let expired_block_index = self
+            .current_slot_index
+            .saturating_sub(max_externalized_slots);
+        let purged_hashes = self.tx_manager.remove_expired(expired_block_index);
+        let pending_values_len_before_purge = self.pending_values.len();
 
         self.pending_values.retain(|value| match value {
             ConsensusValue::TxHash(tx_hash) => !purged_hashes.contains(tx_hash),
+            ConsensusValue::MintConfigTx(mint_config_tx) => {
+                mint_config_tx.prefix.tombstone_block > expired_block_index
+            }
         });
 
         // Drop pending values that are no longer considered valid.
+        let pending_values_len_before_clear_invalid = self.pending_values.len();
         self.pending_values.clear_invalid_values();
 
         log::info!(
             self.logger,
-            "Number of pending values post cleanup: {} ({} expired)",
+            "Number of pending values post cleanup: {} (before purge: {}, before clear_invalid: {})",
             self.pending_values.len(),
-            purged_hashes.len(),
+            pending_values_len_before_purge,
+            pending_values_len_before_clear_invalid,
```

Comment:
Optional/Equivalent: It seems more useful to log the number of purged and expired entries.